File: server-mh/src/Match/Handler/MatchPhases/FinishPhase.ts

import { END_MATCH_TIMEOUT_SECONDS,MATCH_TICK_RATE } from "../Consts";
import { MatchContext } from "../Models/MatchContex";
import { PhaseBase } from "./PhaseBase";

export class FinishPhase extends PhaseBase {

    public override Start(context: MatchContext): void {
        context.state.tickCounter=END_MATCH_TIMEOUT_SECONDS*MATCH_TICK_RATE;
        context.broadcaster.MatchFinish(context.state.winnerList);
    }
    public override Update(context: MatchContext): void {
        context.state.tickCounter--;
        if(context.state.tickCounter<=0)
        {
            context.state.matchEnd=true;
        }
        
    }
}

File: Client-unity/Ludo/Assets/Scripts/OfflineMatchHandler/FinishPhase.cs
using System.Collections.Generic;

namespace Ludo.Offline
{
    public class FinishPhase : PhaseBase
    {
        public override void Start(MatchContext context)
        {
            context.State.TickCounter =
                MatchConstants.EndMatchTimeoutSeconds *
                MatchConstants.MatchTickRate;

            context.CommandHandler.Enqueue(
                new MatchFinishedCommand(
                    new List<PlayerColor>(
                        context.State.WinnerList
                    )
                )
            );
        }

        public override void Update(MatchContext context)
        {
            context.State.TickCounter--;

            if (context.State.TickCounter <= 0)
            {
                context.State.MatchEnd = true;
            }
        }
    }
}